### PR TITLE
Add bias metrics configuration modal

### LINKED ISSUE #1274: Configure bias metric form
### Goal

Implement functionality allowing a user to create and update a bias metric configuration

### Dependency issue

* #1152
* #1269

### Itemized goals

* Implement the configure bias metric modal shown in mocks i and ii, remembering that a user will need to be able to both create a new and update an existing bias metric configuration – therefore design in a flexible way that maximises code reuse.
* Add validation as described on the mock
* Closing the modal via the X or cancel button should behave as described in the mocks
* When the user clicks configure: 
  * Send values to trustyai service to create / update the bias metric config
    * On Success: Close modal, remain on the configure bias metric screen and update the list of configured metrics.
    * On Error: functionality TBD.
* Add logic to the following buttons in the UI to start configuring a new bias metric (i.e. navigate to metrics configuration screen with open modal):
  * Mock iii - ‘Configure’ button on empty state of model bias screen.
  * Mock iv - ‘Configure metric’ button on bias metric configuration list view
  * Mock v - ‘Configure’ button on the info tip of the Deployed models screen.
* Add update logic:
  * On the bias metric configuration list view, the table row actions kebab should have an update action (See mock vi for reference). 
  * When this is clicked the ‘Configure bias metric model’ should be opened and pre-populated with existing values of the ‘bias metric’ for the user to edit.
  * When the user submits, send an update request to the trustyai service.
  * Follow the same post-submit routine as for bias metric creation.

### Aspects continued elsewhere

*  Default violation thresholds will be specific for each “metric type” – to be supplied.

### Mocks

i)
a1dfb24d-b4af-469a-a67a-146a400b6f38
ii)
0fd40fc7-bfdb-4db1-b8e0-605b277ad811
iii)
5fe2434f-9e7e-40e8-96e0-c62076980ce6
iv)
564c2f58-160e-4bfc-8c5d-7ea36d33e0ee
v)
8a086170-e71f-4d3c-9bdd-10e2dfdc9fb3
vi)
78888c5a-a824-42eb-96b2-c4f77c89cd8a

### PR DESCRIPTION
**Pull request by DaoDaoNoCode**
Closes #1274 #1162 

## Description

1. Added a modal that allows you to configure the bias metrics. However, this won't work until https://github.com/trustyai-explainability/trustyai-explainability/pull/186 is merged.

Screenshot 2023-06-06 at 5 01 30 PM

2. Added `Duplicate` and `Delete` for modal metric configuration rows

Screenshot 2023-06-06 at 5 04 38 PM

3. Added popover on the tab when there is no bias metric configured

Screenshot 2023-06-06 at 5 05 59 PM

## How Has This Been Tested?

Go to model -> go to bias tab -> click configure button on the page title -> get into the configuration page -> try to create (not working for now) and delete the configuration

## Test Impact

TBD

## Request review criteria:

- [x] The commits have meaningful messages (squashes happen on merge by the bot).
- [x] Included any necessary screenshots or gifs if it was a UI change.
- [ ] Included tags to the UX team if it was a UI/UX change.
- [ ] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work
- [ ] The developer has added tests or explained why testing cannot be added (unit tests & storybook for related changes)

### CODE REVIEW

**Comment by alexcreasy** [comment]
Testing is all good - I raised a question about the language on the error message on slack. Doing a code review now.

File: frontend/src/api/prometheus/serving.ts [hunk 1]
```diff
@@ -16,7 +20,10 @@ export const useModelServingMetrics = (
   lastUpdateTime: number,
   setLastUpdateTime: (time: number) => void,
 ): {
-  data: Record<RuntimeMetricType, ContextResourceData<PrometheusQueryRangeResultValue>>;
+  data: Record<
+    RuntimeMetricType | InferenceMetricType,
+    ContextResourceData<PrometheusQueryRangeResultValue | PrometheusQueryRangeResponseDataResult>
+  >;
```
[the remark this answers is not in the document]
Comment (DaoDaoNoCode) [reply frontend/src/api/prometheus/serving.ts:26]: It's just a nit, I noticed we are using this for both `InferenceService` and `ServingRuntime`, however, we only set the serving runtime return type here and the compiler didn't complain anything. I just saw this and want to correct it, nothing changed actually :)

File: frontend/src/concepts/explainability/types.ts [hunk 2]
```diff
@@ -36,7 +36,7 @@ export type ExplainabilityAPI = {
 export type BiasMetricConfig = {
   id: string;
   name: string;
-  metricType: MetricTypes;
+  metricType?: MetricTypes;
```
[the remark this answers is not in the document]
Comment (DaoDaoNoCode) [reply frontend/src/concepts/explainability/types.ts:39]: Oh yes you are right, I used this type for creating before, and the required field made me a headache at that time. But then I found we should use `BaseMetricRequest` and separate `metricType`, we don't need to change the type. I forgot to revert the change, will revert it, thanks!

File: frontend/src/api/prometheus/serving.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (alexcreasy) [reply frontend/src/api/prometheus/serving.ts:26]: ahhh ok cool :)

**Comment by alexcreasy** [comment]
lgtm

**Pull request merged**